feat(docs): update Algolia configuration information (#2448)
camunda/camunda-platform-docs: howtos/configuration.md

## Search by Algolia

> Internal search is handled by an integration with Algolia DocSearch.

Reviewer: Is it worth linking to [Algolia DocSearch](https://docsearch.algolia.com/) here?

Author: ```suggestion
Internal search is handled by an integration with [Algolia DocSearch](https://docsearch.algolia.com/).
```

### Configuration

Our Algolia API key, app ID, and index name are configured in `./docusaurus.config.js`.

> Configuration of Algolia's index and crawler can be edited at https://crawler.algolia.com/.

Reviewer: Configuration of Algolia's index and crawler can be edited with the [Algolia Crawler](https://crawler.algolia.com/), or something of the like?

Author: ```suggestion
Configuration of Algolia's index and crawler can be edited with the [Algolia Crawler](https://crawler.algolia.com/).
```

Currently, the only unique change to our configuration is that we specify multiple `pathsToMatch`, to accommodate our multiple docs instances:

```
      pathsToMatch: [
        "https://docs.camunda.io/docs/**",
        "https://docs.camunda.io/optimize/**",
      ],
```

### Troubleshooting

> Algolia's index of our documentation can be explored directly at https://dashboard.algolia.com/.

Reviewer: Take a closer look at [Algolia's index of our documentation](https://dashboard.algolia.com/).

Author: ```suggestion
Take a closer look at [Algolia's index of our documentation](https://dashboard.algolia.com/).
```
